sklearn/cluster/k_means_.py

```diff
@@ -280,7 +283,14 @@ def k_means(X, n_clusters, init='k-means++', precompute_distances='auto',
         raise ValueError('Number of iterations should be a positive number,'
                          ' got %d instead' % max_iter)
 
-    X = as_float_array(X, copy=copy_x)
+    # avoid forcing order when copy_x=False
+    order = "C" if copy_x else None
+    X = check_array(X, accept_sparse='csr', dtype=[np.float64, np.float32],
+                    order=order, copy=copy_x)
+    # verify that the number of samples given is larger than k
+    if _num_samples(X) < n_clusters:
+        raise ValueError("n_samples=%d should be >= n_clusters=%d" % (
```

Comment:
I see that this is cover but I don't see from which test since that you did not add a specific test.
Can you point out which test is raising this?

Reply:
There are quite a few which satisfy this condition, one of the example being https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/cluster/tests/test_k_means.py#L172 called here https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/cluster/tests/test_k_means.py#L178 .

Reply:
Could make the change such that we check the message properly.